src/lib/dlt_user.c

```diff
@@ -493,7 +497,8 @@ DltReturnValue dlt_init(void)
 
     /* Initialize common part of dlt_init()/dlt_init_file() */
     if (dlt_init_common() == DLT_RETURN_ERROR) {
-        dlt_user_init_state = INIT_UNITIALIZED;
+        dlt_user_init_state = INIT_ERROR;
+        dlt_free();
```

Comment:
dlt_free() now needs to be changed to also free resources in INIT_ERROR state. Currently it is using the macro DLT_INITALIZED() which only checks for equality to INIT_DONE.

Reply:
added check for init_error in dlt_free